_posts/2017/2017-12-31-truthiness-in-conditionals.md

```diff
@@ -0,0 +1,57 @@
+---
+title: 'Truthiness in Conditionals'
+description: 'if not not nil?'
+layout: article
+category: ruby
+image:
+  base: '2017/truthiness-in-conditionals'
+  alt: 'Truth'
+  source: 'https://unsplash.com/photos/Aa2h3S3E47k'
+  credit: 'Jarrod Fitzgearlds'
+---
+
+Ruby’s conditional syntax is ‘truthy’, meaning that anything that is `nil` is also considered to be `false` and anything not-`nil` can be considered to be `true`.
+
+
+## Instead of…
+
+...overcomplicating your conditions.
+
+```ruby
+# Example 1
+unless something.nil?
+  # do something
+end
+
+# Example 2
+if !something.nil?
+  # do something
+end
+
+# Example 3
+if !!something
+  # do something
+end
+```
+
+
+## Use…
+
+```ruby
+# Instead of Examples 1,2 & 3
+if something
+  # do something
+end
+```
+
+
+## But why?
+
+Performing a `#nil?` check in a negative conditional, as in the first two examples, is often redundant as any non-`nil` value is `true`. You can remove the `nil?` check and substitute the `unless` for an `if` or remove the `!` and end up with clearer code that means the same thing.
```

Comment:
*Performing a `#nil?` check in a negative conditional, as in the first two examples, is often redundant. This is because any non-`nil` value is truthy on its own. Instead, remove the `#nil?` check and substitute the `unless` for an `if` (for Example 1) or remove the `!` (for Example 2) and end up with clearer code that means the same thing.

Reply:
Even with my suggested changes above...not completely happy with this paragraph...I keep wanting to add more words then I take them out because I think you lose the brevity.

I'll tell you the bits I'm not sure about:

* "...in a negative conditional" — is it clear what you mean here? You mean in a conditional branch that's checking a negative case? Or if you've got 'unless' or '!'-something? Are there any other 'negative' conditionals?

* "as any non-nil value is true" — I'm tempted to add another sentence to explain that the negative statement and the nil check cancel each other out...but then I think it's just getting long..and maybe people do already understand why it's redundant. I also don't think stating the equivalence of non-`nil` and `true` is that clear, hence my suggested edit..but maybe I'm over-thinking the words and how others will read and understand them.